astroquery/eso/core.py

```diff
@@ -695,16 +707,52 @@ def retrieve_data(self, datasets, continuation=False, destination=None):
                     raise RemoteServiceError("There was a remote service "
                                              "error; perhaps the requested "
                                              "file could not be found?")
-            for fileId in root.select('input[name=fileId]'):
-                log.info("Downloading file {0}...".format(fileId.attrs['value'].split()[0]))
-                fileLink = ("http://dataportal.eso.org/dataPortal" +
-                            fileId.attrs['value'].split()[1])
+
+            if with_calib != 'none':
+                # when requested files with calibrations, some javascript is
+                # used to display the files, which prevent retrieving the files
+                # directly. So instead we retrieve the download script provided
+                # in the web page, and use it to extract the list of files.
+                # The benefit of this is also that in the download script the
+                # list of files is de-duplicated, whereas on the web page the
+                # calibration files would be duplicated for each exposure.
+                try:
+                    link = root.select('a[href$=/script]')[0]
+                    if 'downloadRequest' not in link.text:
+                        # Make sure that we found the correct link
+                        raise Exception
+                    href = link.attrs['href']
+                    script = self._request("GET", href, cache=False)
+                    fileLinks = re.findall(
+                        r'"(https://dataportal.eso.org/dataPortal/api/requests/.*)"',
+                        script.content.decode('utf8'))
+                    # links with api/ do not work, wtf ???
+                    fileLinks = [
+                        f.replace('https://dataportal.eso.org/dataPortal/api/requests',
+                                  'https://dataportal.eso.org/dataPortal/requests')
```

Comment:
This, I don't understand.. the script contains links with "api/", which should work obviously (and works when executing the script), but doesn't work as a direct http request. So I remove "api/" and get the same urls as with the web interface.